src/client/ns_turn_msg.c

```diff
@@ -248,24 +248,24 @@ int stun_produce_integrity_key_str(const uint8_t *uname, const uint8_t *realm, c
 		if (FIPS_mode()) {
 			EVP_MD_CTX_set_flags(&ctx,EVP_MD_CTX_FLAG_NON_FIPS_ALLOW);
 		}
-#endif
+#endif // defined EVP_MD_CTX_FLAG_NON_FIPS_ALLOW && !defined(LIBRESSL_VERSION_NUMBER)
 		EVP_DigestInit_ex(&ctx,EVP_md5(), NULL);
 		EVP_DigestUpdate(&ctx,str,strl);
 		EVP_DigestFinal(&ctx,key,&keylen);
 		EVP_MD_CTX_cleanup(&ctx);
-#else
+#else // OPENSSL_VERSION_NUMBER < 0x10100000L
 		unsigned int keylen = 0;
 		EVP_MD_CTX *ctx = EVP_MD_CTX_new();
-#if defined EVP_MD_CTX_FLAG_NON_FIPS_ALLOW && ! defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
I think with this change we are not disabling FIPS anymore for openssl versions > 0x10100000L and < 0x30000000L.  Is that correct?

Reply:
I do not know much about FIPS but I could find following
- Only 1.0.2 and 3.0.2 are FIPS certified (and some even older 0.9.8)
- All 3.0 contain FIPS code (but only 3.0.2 is certified)
So my new code is supposed to do just that (enable for 1.0.2 in one branch and for 3.0 for another)

Reply:
I do not know much about FIPS but I could find following
- Only 1.0.2 and 3.0.2 are FIPS certified (and even older 0.9.8)
- All 3.0 contain FIPS code (but only 3.0.2 is certified)
So my new code is supposed to do just that (enable for 1.0.2 in one branch and for 3.0 for another) - but yes I see how there is a new gap